kotlin-inject-compiler-kapt/src/main/kotlin/me/tatarka/inject/compiler/kapt/ModelAst.kt

```diff
@@ -232,20 +231,34 @@ private class ModelAstClass(
 
     override val primaryConstructor: AstConstructor?
         get() {
-            return ElementFilter.constructorsIn(element.enclosedElements).mapNotNull { constructor ->
-                //TODO: not sure how to match constructors
+            kmClass ?: return null
```

Comment:
On a side note, have you considered adding ktlint and/or detekt and running checks and tests on each pull request? Not that it would catch this particular instance, ktlint doesn't have an opinion on this specific language construct, but it could save some time with future reviews and provide an opinionated code style to follow.

Reply:
👍 Yeah, I should do that.